lib/portage/dbapi/bintree.py

```diff
@@ -1628,8 +1789,16 @@ def _allocate_filename_multi(self, cpv):
 		pf = catsplit(cpv)[1]
 		build_id = max_build_id + 1
 
+		binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
```

Comment:
`_allocate_filename_multi` is similar to `_allocate_filename`, `getname` and `_pkgindex_entry`, we should not assume a constant binpkg_format here.

Reply:
Maybe `cpv._metadata['BINPKG_FORMAT']` can solve most or all of these.